### PR TITLE
Create generate_dataset.py

### PR DESCRIPTION
**Pull request by ClemHubble**
This is the code to generate the npz datasets from text files.

### CODE REVIEW

File: generate_dataset.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+    G = nx.Graph()
+    for (src, tgt, dist) in zip(remapped_edges[0], remapped_edges[1], distances):
+        G.add_edge(int(src), int(tgt), weight=dist)
+
+    srcs = []
+    tars = []
+    lengths = []
+    node_indices = list(G.nodes)
+    np.random.seed(42)
```
Comment (chens5) [inline generate_dataset.py]: Take out random seed

File: generate_dataset.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    N = int(50000 / 0.8)
```
Comment (chens5) [inline generate_dataset.py]: Either set as constants or explain in a comment what these are. I believe these are dataset sizes but make sure that we're using 50,000 pairs of (src, tar, shortest path distances) for training.

File: generate_dataset.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
+import numpy as np
+import networkx as nx
+from sklearn.model_selection import train_test_split
+import argparse
+
+def generate_dataset(vertex_file, edge_file, output_prefix):
```
Comment (chens5) [inline generate_dataset.py]: Name this file: generate_simplified_mesh_dataset.py. We want to use this script to take a particular terrain DEM, specify the simplification level, and then generate the simplified mesh in the `.npz` format for training.

**Review by chens5 — CHANGES_REQUESTED** [review]
